File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.js
var replaceElements = function (arr) {
    let max_element = -1
    for (let i = arr.length - 1; i >= 0; i--) {
        [arr[i], max_element] = [max_element, Math.max(arr[i], max_element)]
    }
    return arr
};

File: 1231-replace-elements-with-greatest-element-on-right-side/replace-elements-with-greatest-element-on-right-side.cpp
class Solution {
public:
    vector<int> replaceElements(vector<int>& arr) {
        int n = arr.size();
        int max_element = -1;
        for (int i = n - 1; i >= 0; i--) {
            int current_val = arr[i];
            arr[i] = max_element;
            max_element = max(max_element, current_val);
        }
        return arr;
    }
};